src/api/java/baritone/api/utils/ExampleBaritoneControl.java

```diff
@@ -365,8 +379,51 @@ public boolean isInGoal(int x, int y, int z) {
             logDirect("okay");
             return true;
         }
-        if (msg.equals("farm")) {
-            baritone.getFarmProcess().farm();
+        if (msg.startsWith("farm")) {
+            String rest = msg.substring(4).trim();
+            String[] params = rest.split(" ");
+            int[] args = new int[]{-1, 0, 0, 0};
+            if (params.length > 0) {
+                args[0] = Integer.parseInt(params[0]);
+            }
+            try {
+                switch (params.length) {
```

Comment:
this is horrifying.

Reply:
Should I have `args[0] = Integer.parseInt(params[0]);` in every case from 1 to 4 instead?

Reply:
no you're just really overthinking it and this is wild complicated for no reason

imo there should be a farm range setting. to change the farm range, change that setting.

the farm center should be your playerfeet, or you pass x y z as args. farm should take 0 arguments or 3.

Reply:
that's stupid. while using this for myself I never had any use for setting the coordinates. But I constantly need a different range of settings. 

I just realized that one problem with the coordinates function is very easy to fix. If you are too far away from the target center, it will not work. to fix this is to simply use the `goto`.
so the command `#farm 10 x y z` will be translated to `#goto x y z #farm 10`.

Reply:
truly horrifying

Reply:
then make those separate settings. this is too much complexity in farm lol.

goto already has the logic for parsing this.

i'm fine if you want to do `farm radius` = farm at that radius and `farm` = farm with no radius. in that case, remove the farm radius setting.

Reply:
soooo...
`farm` that works just like the original and
`farmrange` that have a range setting.

What we/I can do... is:
`farmrange i` farm within the range of `i` from player feet.
`farmrange  x1 y1 z1 x2 y2 z2` or `farmrange  x1 z1 x2 z2` to specify an area.
`farmrange i shape` farm within the range of `i` from player feet with a specific shape like a rectangle.
`farmthis` auto-set aria of the current position and farm it.